src/main/java/org/elasticsearch/index/query/CommonTermsQueryBuilder.java

```diff
@@ -50,40 +69,62 @@
         OR, AND
     }
 
-    private final String name;
+    private String fieldName;
 
-    private final Object text;
+    private String text;
```

Comment:
we can leave final fields (at least the mandatory ones) now that we implement Writeable instead of Streamable, then we can (almost) drop default constructors (see #11344 for more details)